### PR TITLE
Extract NLU module to make it testable.

### PR DESCRIPTION
**Pull request by LittleYenMin**
#1

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
+*$py.class
+
+# C extensions
+*.so
```
Comment (dannypsnl) [inline .gitignore]: I suggest don't use any template for git ignore, which who make you later forgot what you add into here, manually remove those you don't want would be better

File: app.py [hunk 2]
```diff
@@ -13,35 +13,35 @@
...
         print("Invalid signature. Please check your channel access token/channel secret.")
         abort(400)
-
     return 'OK'
 
+
 @handler.add(MessageEvent, message=TextMessage)
 def handle_message(event):
-    text = jieba.cut(event.message.text)
+    text = nlu.execute(event.message.text)
```
Comment (dannypsnl) [inline app.py]: execute absolutely not a good name here, rename to something like response_message

File: tests/test_nlu.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import pytest
+import nlu
+
+@pytest.fixture()
+def text():
+    return '我想查曠課'
+
+def test_nlu_execute(text):
+    assert nlu.execute(text) == '我想查, 曠課'
```
Comment (dannypsnl) [inline tests/test_nlu.py]: Do not test external lib, at here what you should test should be can the NLU returns correct command you want to be run or what, if we can't do that right now then please remove the test, its annoying and useless at here

File: .gitignore (hunk 1 above)
Comment (LittleYenMin) [reply .gitignore]: you are right.

File: app.py (hunk 2 above)
Comment (LittleYenMin) [reply app.py]: you're right again.

File: tests/test_nlu.py (hunk 3 above)
Comment (LittleYenMin) [reply tests/test_nlu.py]: agree

**Comment by dannypsnl** [comment]
@LittleYenMin BTW, don't mark resolve, it should be marked by me so that I can sure it's no problem and no need the open it again to unmark it.

**Comment by LittleYenMin** [comment]
sorry, my bed 我的床

**Comment by dannypsnl** [comment]
No mind, I also just knew that a few days.

**Pull request merged**